platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/cache/VirtualNodeCache.java

```diff
@@ -611,8 +609,9 @@ public void deleteLeaf(VirtualLeafRecord<K, V> leaf) {
      */
     public void clearLeafPath(long path) {
         throwIfLeafImmutable();
-        // Note: this marks the mutation as deleted, in addition to clearing the value of the mutation
+        // Note: this marks the mutations as deleted, in addition to clearing the value of the mutation
         updatePaths(null, path, pathToDirtyLeafIndex, dirtyLeafPaths);
+        updatePaths(null, path, pathToDirtyHashIndex, dirtyHashes);
```

Comment:
There should be no hash for a mutable leaf yet. Is it needed?

Reply:
Yes, it's redundant. I'll remove it.

Reply:
In fact, this test after the removal: `VirtualMapHashingTest#internalNodeSynchronization`. Could you please confirm that it's synthetic and doesn't correspond to the real life scenario?

Reply:
See my comment below regarding the failing test.

Reply:
This is a tricky case. We used to delete a leaf that contained a hash. Now, since hashes are separated from leaves, it seems logical that we need to clear the hash from a separate map. However, there is no new hash yet in the current version but, more importantly, we never cared about clearing parent's hashes, which are also invalidated by changes to the leaf. The hashing algorithm would work correctly from dirty leaves and doesn't need dirty leaf hashes to be invalidated either. Still, this is an implicit dependency on hasher's implementation.